recurly/base_client.py

```diff
@@ -66,9 +71,16 @@ def _make_request(self, method, path, body, **options):
             if "params" in options:
                 path += "?" + self._url_encode(options["params"])
 
+            # Every __conn must have getresponse() called on it successfully
+            # or it must be thrown away
+            if self.__needs_reset:
+                self._create_conn()
```

Comment:
It seems like to recreate the entire connection every call might be wasteful for the problem. However i'm not sure what the result of this is. Does it also recreate the underlying connection or just the connection object? Is there a more precise way to detect when the connection needs to be reset?

Reply:
`needs_reset` is toggled back to False after a successful Response() (see below). So this path is only ever triggered when you had an exception that prevented the toggling.

Reply:
seems simple enough!

Reply:
Another way of looking at it: if you were creating a new Recurly object for each API call you avoided this bug but you made a lot of redundant connection objects and Recurly instances. If you didn't make a new Recurly object each time your Recurly instance would occasionally break when the unhandled exception was raised below. But with this change you're reusing a connection object if there is no exception raised and re-creating one when there is an exception / it's in the broken state.